stores.js

```diff
@@ -192,6 +196,15 @@ class MutableStore extends Hypersign {
     const queryStream = dht.update('mutable-store', key, {
       value, signature, seq, salt
     })
+
+    queryStream.once('warning', (err, proof) => {
+      if (proof) {
+        const { value, signature, seq, salt } = proof
+        const msg = this.signable(value, { salt, seq })
+        const verified = verify(signature, msg, key)
```

Comment:
Based on the error message you have to here validate that error is indeed what the remote says.

Ie. if the remote says ERR_INVALID_SEQ, you have to perform the local version of the check performed here to ensure the remote is not lying, https://github.com/hyperswarm/dht/pull/26/files#diff-1071e9eefc1ddc6530576ccf7bc9adeaR241

Same for the other error

Reply:
Otherwise this is vuln to a preimage attack (ie send some valid proof, but attack the error message itself)